test/opt/upgrade_memory_model_test.cpp

```diff
@@ -1430,6 +1429,288 @@ OpFunctionEnd
   SinglePassRunAndMatch<opt::UpgradeMemoryModel>(text, true);
 }
 
-#endif
+TEST_F(UpgradeMemoryModelTest, UpgradeModfNoFlags) {
+  const std::string text = R"(
+; CHECK: [[float:%\w+]] = OpTypeFloat 32
+; CHECK: [[float_0:%\w+]] = OpConstant [[float]] 0
+; CHECK: [[ptr:%\w+]] = OpTypePointer StorageBuffer [[float]]
+; CHECK: [[var:%\w+]] = OpVariable [[ptr]] StorageBuffer
+; CHECK: [[struct:%\w+]] = OpTypeStruct [[float]] [[float]]
+; CHECK: [[modfstruct:%\w+]] = OpExtInst [[struct]] {{%\w+}} ModfStruct [[float_0]]
+; CHECK: [[ex0:%\w+]] = OpCompositeExtract [[float]] [[modfstruct]] 0
+; CHECK: [[ex1:%\w+]] = OpCompositeExtract [[float]] [[modfstruct]] 1
+; CHECK: OpStore [[var]] [[ex1]]
+; CHECK-NOT: NonPrivatePointerKHR
```

Comment:
Why shouldn't NonPrivatePointerKHR be there?  It seems it's the only safe thing to do, barring specific knowledge (and the compiler doesn't have that knowledge)

Reply:
The pointer is not annotated with Coherent. All the upgrades currently require either coherent or being Workgroup storage class to get NonPrivate.

Do you think the default should be NonPrivate operations? That's a change that should apply widely to this pass.

Reply:
Makes sense. Thanks.